components/director/internal/domain/scenarioassignment/repository.go

```diff
@@ -15,21 +20,42 @@ var tenantColumn = "tenant_id"
 func NewRepository(conv EntityConverter) *repository {
 	return &repository{
 		creator: repo.NewCreator(tableName, columns),
+		lister:  repo.NewLister(tableName, tenantColumn, columns),
 		conv:    conv,
 	}
 }
 
 type repository struct {
 	creator repo.Creator
+	lister  repo.Lister
 	conv    EntityConverter
 }
 
 //go:generate mockery -name=EntityConverter -output=automock -outpkg=automock -case=underscore
 type EntityConverter interface {
 	ToEntity(assignment model.AutomaticScenarioAssignment) Entity
+	FromEntity(assignment Entity) *model.AutomaticScenarioAssignment
 }
 
 func (r *repository) Create(ctx context.Context, model model.AutomaticScenarioAssignment) error {
 	entity := r.conv.ToEntity(model)
 	return r.creator.Create(ctx, entity)
 }
+
+func (r *repository) GetForSelector(ctx context.Context, in model.LabelSelector, tenant string) ([]*model.AutomaticScenarioAssignment, error) {
+	var out EntityCollection
+
+	conditionKey := fmt.Sprintf("selector_key = %s", pq.QuoteLiteral(in.Key))
```

Comment:
Please make the `selector_key` and `selector_value` variables and reuse them also in `columns` collection, similarly to `tenantColumn`